optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/roster/EmployeeRosterViewPanel.java

```diff
@@ -94,6 +94,10 @@ protected void refreshTable() {
             @Override
             public void onSuccess(EmployeeRosterView employeeRosterView) {
                 EmployeeRosterViewPanel.this.employeeRosterView = employeeRosterView;
+                if (null == employeeRosterView) {
```

Comment:
This might be null due to other reasons.
I guess we can't detect it on the server side? as it happens during the json marshalling?

Reply:
Yeah; none of our server-side code is touched unfortunately. The only way to detect it server side that I can think of is to expose a new REST endpoint, serverup, which returns a JSON with a boolean "isUp", which is true iff the server is working property. To calculate if the server is running property, we can try {...}catch(Exception e){...} the call which throws the exception. But I think that kinda pointless. If there a way for the server to process the html files before severing them, then we can do a pre-emptive check using the above method, and return an error HTML page if it is setup incorrectly.

Reply:
Agreed. Let's add some doubt in the message to the user (the word "maybe"), in case it's not failing due to the wrong WildFly version, for example let's change
  "You are using the wrong version of Wildfly. Please use Wildfly 10.1.0.Final"
into
  "The server returned an empty result which is impossible.\nMaybe the server is using the wrong WildFly version. Try using WildFly 10.1.0.Final."